bench/mitsuba3/optimized/regular.ll

```diff
@@ -3754,34 +3754,36 @@ common.resume:                                    ; preds = %13, %24, %111, %33,
   %44 = fmul contract double %41, 5.000000e-01
   br label %45
 
-45:                                               ; preds = %.lr.ph, %70
-  %46 = phi float [ %42, %.lr.ph ], [ %..i, %70 ]
-  %.0113147 = phi ptr [ %1, %.lr.ph ], [ %51, %70 ]
-  %.0115146 = phi i64 [ 0, %.lr.ph ], [ %77, %70 ]
-  %.0116145 = phi double [ 0.000000e+00, %.lr.ph ], [ %57, %70 ]
-  %47 = phi i32 [ -1, %.lr.ph ], [ %76, %70 ]
-  %48 = phi i32 [ -1, %.lr.ph ], [ %74, %70 ]
-  %49 = load float, ptr %.0113147, align 4
-  %50 = fpext float %49 to double
-  %51 = getelementptr inbounds nuw i8, ptr %.0113147, i64 4
-  %52 = load float, ptr %51, align 4
-  %53 = fpext float %52 to double
-  %54 = fadd contract double %50, %53
-  %55 = fmul contract double %44, %54
-  %56 = fcmp contract olt float %46, %52
-  %..i = select contract i1 %56, float %52, float %46
+45:                                               ; preds = %.lr.ph, %77
+  %46 = phi float [ %42, %.lr.ph ], [ %..i, %77 ]
+  %.0113151 = phi ptr [ %1, %.lr.ph ], [ %53, %77 ]
+  %.0115150 = phi i64 [ 0, %.lr.ph ], [ %82, %77 ]
+  %.0116149 = phi double [ 0.000000e+00, %.lr.ph ], [ %57, %77 ]
+  %47 = phi i32 [ -1, %.lr.ph ], [ %81, %77 ]
+  %48 = phi i32 [ -1, %.lr.ph ], [ %80, %77 ]
+  %49 = phi i32 [ -1, %.lr.ph ], [ %79, %77 ]
+  %50 = phi i32 [ -1, %.lr.ph ], [ %78, %77 ]
```

Comment:
Regression: Extra phi nodes. Probably 49/50 are same as 47/48, but we're currently not good at deduplicating phis with circular deps.